lib/msf/base/simple/exploit.rb

```diff
@@ -148,11 +148,15 @@ def self.exploit_simple(oexploit, opts, &block)
 
       # Propagate this back to the caller for console mgmt
       oexploit.job_id = exploit.job_id
+      oexploit.fail_reason = exploit.fail_reason
+      oexploit.fail_detail = exploit.fail_detail
```

Comment:
I'm not sure on the lore of why exploits additionally have their own separate lossy error information fields, or why `check_simple` doesn't use replicants.